### ISSUE TITLE
Support Android Vector Drawables

### ISSUE THREAD

**Issue by zugaldia** [mapbox/maki#247]
The goal of this ticket is to bring native support to Maki on Android using Vector Drawables. This is both useful for the current Mapbox SDK and for the upcoming mobile navigation SDK.

## Background

Android [Vector Drawables](http://developer.android.com/training/material/drawables.html#VectorDrawables) are XML files that contain an `android:pathData` element which follows the SVG Path reference. This means that although Android Vector Drawables are not pure SVG files, the transformation between the two should be straightforward (at least in theory).

Consuming a Vector Drawable in Android hasn't been historically easy to do though, because it was only supported by recent versions of Android. Instead, to remain backwards compatible, developers had to create different sizes of the same bitmap file and add them to their project. This has changed recently with the introduction of the Android Support Library 23.2 that brings support for API levels back to 11, in practice, all API levels supported by the Mapbox Android SDK (15+).

Related ticket: https://github.com/mapbox/mapbox-gl-native/issues/4112

## Next steps

- [ ] Test with [Android's Asset Studio](http://developer.android.com/tools/help/vector-asset-studio.html) during next Maki release
- [ ] Automate import process
- [ ] Evaluate packaging Maki with Mapbox Android SDK
- [ ] Evaluate packaging Maki with Mapbox Android Navigation SDK

/cc: @samanpwbb @tobrun @bleege 

_Note: Ticket moved from original in https://github.com/mapbox/maki-2/issues/43_

**Comment by zugaldia** [comment]
I've managed to get some partial success in https://github.com/mapbox/maki-2/tree/43-android:

![video](https://cloud.githubusercontent.com/assets/6964/13986571/7cd0a376-f0d9-11e5-97af-c8828598fb7a.gif)

For the icons not loading, I'm getting the following stack:

```
W/PathParser: Points are too far apart 4.000002598526946
W/PathParser: Points are too far apart 4.0180006809234925
W/PathParser: Points are too far apart 4.014397245026217
W/PathParser: Points are too far apart 4.032800433349621
W/PathParser: Points are too far apart 4.000401887512453
...
```

Source for the `PathParser` is here: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/util/PathParser.java#579

Next is to figure out if I need to some extra path math to bring SVG to Android (likely seeing what the Android Vector Asset Studio does). For now, I'm just copying the path data verbatim.

**Comment by zugaldia** [comment]
Just checked, and the same thing happens with the 11px version of the icons.

**Comment by zugaldia** [comment]
There's definitely some extra conversion going on in Android Vector Asset Studio. For example, this is the original path for zoo-11 (that doesn't load currently):

```
M-286,197c-0.9,0-1.5,0.5-1.8,1.1c0,0-0.7,0.9-1.2,0.9h-2c-0.5,0-1,0.5-1,0.5l-2,2v0.5h0.5l1.2-1.2l0.2,0.2  c0,0-0.7,1.3-0.7,2.6c0,0.5,0.5,0.5,0.5,0.5h0.5c0,0,0.2,0,0-0.2l-0.2-0.2c0-0.5,0.6-1,1-1.5c0,0,0,0.6,0,1.2c0,0.3,0.2,0.8,0.5,0.8  h0.5c0,0,0.2,0,0-0.2l-0.2-0.2c-0.2-0.4,0.5-1.5,0.5-1.5h1.8l0.5,1.5c0.2,0.5,0.5,0.5,0.5,0.5h1c0.2,0,0.7-0.2,0.2-0.5l-0.2-0.2V202  v0c1,0,1.1-1.3,1.5-1.7l0.7-0.1c0.2,0,0.8-0.2,0.8-0.8V199l-1.2-1l0,0C-284.6,197.4-285.3,197-286,197z
```

And this is after Android Vector Asset Studio imports it (this one does work):

```
M8,2c-0.9,0 -1.5,0.5 -1.8,1.1c0,0 -0.7,0.9 -1.2,0.9h-2c-0.5,0 -1,0.5 -1,0.5l-2,2v0.5h0.5l1.2,-1.2l0.2,0.2c0,0 -0.7,1.3 -0.7,2.6c0,0.5 0.5,0.5 0.5,0.5h0.5c0,0 0.2,0 0,-0.2l-0.2,-0.2c0,-0.5 0.6,-1 1,-1.5c0,0 0,0.6 0,1.2c0,0.3 0.2,0.8 0.5,0.8h0.5c0,0 0.2,0 0,-0.2l-0.2,-0.2c-0.2,-0.4 0.5,-1.5 0.5,-1.5h1.8l0.5,1.5c0.2,0.5 0.5,0.5 0.5,0.5h1c0.2,0 0.7,-0.2 0.2,-0.5l-0.2,-0.2L8.1,7v0c1,0 1.1,-1.3 1.5,-1.7l0.7,-0.1c0.2,0 0.8,-0.2 0.8,-0.8L11.1,4l-1.2,-1l0,0C9.4,2.4 8.7,2 8,2z
```

**Comment by zugaldia** [comment]
@samanpwbb After https://github.com/mapbox/mapbox-java/pull/48 landed, current Maki icons are now included in the library with an example app for Android developers on how to use them. How could we advertise this from the Maki side? Maybe a note in `README`?

**Comment by samanpwbb** [comment]
@zugaldia yeah, let's definitely get a note in the read me here. We could even mention that maki is available in our android sdk on the maki website.

**Comment by zugaldia** [comment]
@samanpwbb I like that, adding a note in the `README` + website looks good to me.

@Lndsay Could you add this to our mobile comms blog backlog too? I think we could have a nice post in the future showcasing how to use Maki icons on Android.

@cammace could you coordinate with @samanpwbb here? For now we could simply add an `Android` (or `Mobile`) subsection in the `README`, link to MAS, and maybe to the [`MakiIconsActivity`](https://github.com/mapbox/mapbox-java/blob/master/libandroid/app/src/main/java/com/mapbox/services/android/testapp/icons/MakiIconsActivity.java) (or extract a new sample).

**Comment by cammace** [comment]
@zugaldia I'll create an example on android-sdk/examples and then we can link it in the readme. issue for tracking example mapbox/android-sdk#124

**Comment by Lndsay** [comment]
> Could you add this to our mobile comms blog backlog too?

Done
